### ISSUE TITLE
Encountered a problem with binary classification in Segmenter

### ISSUE THREAD

**Issue by plainzzj** [open-mmlab/mmsegmentation#2901]
Hello!

I tried using a Segmentor to complete a binary classification task, but encountered an issue with a foreground training result of 0 with a pixel value of 1:

![image](https://user-images.githubusercontent.com/93184749/231947457-d1b45dac-5e8f-4f53-9ea7-d0a4e2d640fd.png)

Previously, I have used a series of networks such as Segformer, DeepLab v3, CCNet, etc., but have not encountered any of the above issues;
I used chase_db1 again to confirm and obtained the same result:

![image](https://user-images.githubusercontent.com/93184749/231947961-2db07ee6-ee69-42f2-aab0-42969d3daff0.png)
The version I am using is v0.27.0
Hope to get help! thank you!
`checkpoint = 'https://download.openmmlab.com/mmsegmentation/v0.5/pretrain/segmenter/vit_base_p16_384_20220308-96dfe169.pth'
backbone_norm_cfg = dict(type='LN', eps=1e-06, requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained=
    'https://download.openmmlab.com/mmsegmentation/v0.5/pretrain/segmenter/vit_base_p16_384_20220308-96dfe169.pth',
    backbone=dict(
        type='VisionTransformer',
        img_size=(512, 512),
        patch_size=16,
        in_channels=3,
        embed_dims=768,
        num_layers=12,
        num_heads=12,
        drop_path_rate=0.1,
        attn_drop_rate=0.0,
        drop_rate=0.0,
        final_norm=True,
        norm_cfg=dict(type='LN', eps=1e-06, requires_grad=True),
        with_cls_token=True,
        interpolate_mode='bicubic'),
    decode_head=dict(
        type='SegmenterMaskTransformerHead',
        in_channels=768,
        channels=768,
        num_classes=2,  # Change to 2;
        num_layers=2,
        num_heads=12,
        embed_dims=768,
        dropout_ratio=0.0,
        #  SegmenterMaskTransformerHead: __init__() got an unexpected keyword argument 'out_channel'
        # out_channel=2,
        ignore_index=255,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),  # use_sigmoid = False
    test_cfg=dict(mode='slide', crop_size=(512, 512), stride=(480, 480)))
dataset_type = 'AppleDataset'
data_root = 'data/zzj_apple/MinneApple'
img_norm_cfg = dict(
    mean=[127.5, 127.5, 127.5], std=[127.5, 127.5, 127.5], to_rgb=True)
crop_size = (512, 512)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=False),  # Change to false;
    dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(
        type='Normalize',
        mean=[127.5, 127.5, 127.5],
        std=[127.5, 127.5, 127.5],
        to_rgb=True),
    dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(2048, 512),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[127.5, 127.5, 127.5],
                std=[127.5, 127.5, 127.5],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=2,
    train=dict(
        type='AppleDataset',
        data_root='data/zzj_apple/MinneApple',
        img_dir='detection/train/images',
        ann_dir='detection/train/masks',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', reduce_zero_label=False),  # Change to false;
            dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[127.5, 127.5, 127.5],
                std=[127.5, 127.5, 127.5],
                to_rgb=True),
            dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='AppleDataset',
        data_root='data/zzj_apple/MinneApple',
        img_dir='test_data/segmentation/images',
        ann_dir='test_data/segmentation/masks',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(2048, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[127.5, 127.5, 127.5],
                        std=[127.5, 127.5, 127.5],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='AppleDataset',
        data_root='data/zzj_apple/MinneApple',
        img_dir='test_data/segmentation/images',
        ann_dir='test_data/segmentation/masks',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(2048, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[127.5, 127.5, 127.5],
                        std=[127.5, 127.5, 127.5],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
log_config = dict(
    interval=50, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
cudnn_benchmark = True
optimizer = dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=0.0)
optimizer_config = dict()
lr_config = dict(policy='poly', power=0.9, min_lr=0.0001, by_epoch=False)
runner = dict(type='IterBasedRunner', max_iters=32000)
checkpoint_config = dict(by_epoch=False, interval=1000)
evaluation = dict(interval=500, metric='All')
work_dir = './work_dirs/apple_segmentor_full'
gpu_ids = [0]
auto_resume = False
`

**Comment by plainzzj** [comment]
我看到了对应的版本更新，现在我想使用新的版本1.x，如何让新旧两个版本共存呢，按照教程的步骤，我在git clone的时候遇到了目标路径'mmsegmentation'已经存在，并不是一个空目录的错误信息，我该怎样操作？

I have seen the corresponding version update, and now I want to use the new version 1. x. How can I make the new and old versions coexist? Following the tutorial steps, I encountered an error message while Git Clone that the target path 'mmsegmentation' already exists and is not an empty directory. What should I do？

**Comment by viibridges** [comment]
i got exactly the same problem. similar issues about **segmenter** training were mentioned over and over in the issue history (#1351, #2757, #2117, etc.), but had never been fully solved (some were closed because of inactivity, people may just loss their patience with **segmenter**)
**segformer** works well for and always give satisfying results. there must something wrong with either the implementation of the code or incorrect configurations i wrote. i would still be happy to see some one find out the bug.

**Comment by chefkrym** [comment]
> i got exactly the same problem. similar issues about **segmenter** training were mentioned over and over in the issue history (#1351, #2757, #2117, etc.), but had never been fully solved (some were closed because of inactivity, people may just loss their patience with **segmenter**) **segformer** works well for and always give satisfying results. there must something wrong with either the implementation of the code or incorrect configurations i wrote. i would still be happy to see some one find out the bug.

Hey, I'm totally new and wanted to do a binary segmentation on my own dataset using Segformer. Could you be so kind as to share your working code please? Thank you so much :)

**Comment by chefkrym** [comment]
> Hey @viibridges , I'm totally new and wanted to do a binary segmentation on my own dataset using Segformer. Could you be so kind as to share your working code please? Thank you so much :)

**Comment by viibridges** [comment]
> > i got exactly the same problem. similar issues about **segmenter** training were mentioned over and over in the issue history (#1351, #2757, #2117, etc.), but had never been fully solved (some were closed because of inactivity, people may just loss their patience with **segmenter**) **segformer** works well for and always give satisfying results. there must something wrong with either the implementation of the code or incorrect configurations i wrote. i would still be happy to see some one find out the bug.
> 
> Hey, I'm totally new and wanted to do a binary segmentation on my own dataset using Segformer. Could you be so kind as to share your working code please? Thank you so much :)

hi @chefkrym , the secret is not feeding any inputs to the prompt encoder. you are free to browse my code in this link (https://github.com/viibridges/sam-finetuning), may it help you.

**Comment by viibridges** [comment]
> > Hey @viibridges , I'm totally new and wanted to do a binary segmentation on my own dataset using Segformer. Could you be so kind as to share your working code please? Thank you so much :)

i am sorry @chefkrym i can't, i don't own the copyright of my code because it belongs to the company i am working for. but it's easy to make your customized dataset to work with mmsegmentation, as long as you follow the guide, and i encourage you to do that because it's beneficial.
a set of tips:
1. curate your data to the format that segformer training needed
2. pick one of the segformer config files (any one you would like)
3. run train command with your segformer config as inputs

**Comment by chefkrym** [comment]
> > > Hey @viibridges , I'm totally new and wanted to do a binary segmentation on my own dataset using Segformer. Could you be so kind as to share your working code please? Thank you so much :)
> 
> i am sorry @chefkrym i can't, i don't own the copyright of my code because it belongs to the company i am working for. but it's easy to make your customized dataset to work with mmsegmentation, as long as you follow the guide, and i encourage you to do that because it's beneficial. a set of tips:
> 
> 1. curate your data to the format that segformer training needed
> 2. pick one of the segformer config files (any one you would like)
> 3. run train command with your segformer config as inputs

Thanks so much @viibridges :)
